File: 2024/Node/adventSrc/day14/robot.mjs
export class Robot { 
    constructor(x, y, v, maxX, maxY) {
        this.x = x;
        this.y = y;
        this.v = v;
        this.maxX = maxX || 101;
        this.maxY = maxY || 103;
    }
    move() {
        this.x += this.v.x;
        this.y += this.v.y;
        if(this.x < 0) {
            this.x = this.maxX + this.x;
        } else if(this.x >= this.maxX) {
            this.x = this.x - this.maxX; 
        }
        if(this.y < 0) {
            this.y = this.maxY + this.y;
        } else if(this.y >= this.maxY) {
            this.y = this.y - this.maxY;
        }
    }
    printNextMove() {
        let nx = this.x + this.v.x;
        let ny = this.y + this.v.y;
        if(nx < 0) {
            nx = this.maxX + nx;
        } else if(nx >= this.maxX) {
            nx = nx - this.maxX; 
        }
        if(ny < 0) {
            ny = this.maxY + ny;
        } else if(ny >= this.maxY) {
            ny = ny - this.maxY;
        }
        console.log(`Robot(${this.x},${this.y},(${this.v.x},${this.v.y}))`, '->', `Robot(${nx},${ny})`);
    }   
    toString() {
        return `Robot(${this.x},${this.y},${this.v})`;
    }
}

File: 2024/dotNet/Day14WinFormsApp/Model/Robot.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace aocDotnetApp
{
    internal class Robot
    {
        public int X { get; set; }
        public int Y { get; set; }
        public int Dx { get; }
        public int Dy { get; }
        public int MaxX { get; }
        public int MaxY { get; }

        public Robot(int x, int y, int dx, int dy, int maxX = 101, int maxY = 103)
        {
            X = x;
            Y = y;
            Dx = dx;
            Dy = dy;
            MaxX = maxX;
            MaxY = maxY;
        }

        public void Move()
        {
            this.X += this.Dx;
            this.Y += this.Dy;
            if (this.X < 0)
            {
                this.X += this.MaxX;
            }
            if (this.X >= this.MaxX)
            {
                this.X -= this.MaxX;
            }
            if (this.Y < 0)
            {
                this.Y += this.MaxY;
            }
            if (this.Y >= this.MaxY)
            {
                this.Y -= this.MaxY;
            }
        }

        public override string ToString()
        {
            return "Robot at (" + this.X + ", " + this.Y + ")";
        }
    }
}
